src/AspNet.Security.OAuth.Instagram/InstagramAuthenticationHandler.cs

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using AspNet.Security.OAuth.Extensions;
+using Microsoft.AspNet.Authentication;
+using Microsoft.AspNet.Authentication.OAuth;
+using Microsoft.AspNet.Http.Authentication;
+using Microsoft.Extensions.Internal;
+using Newtonsoft.Json.Linq;
+
+namespace AspNet.Security.OAuth.Instagram {
+    public class InstagramAuthenticationHandler : OAuthHandler<InstagramAuthenticationOptions> {
+        public InstagramAuthenticationHandler([NotNull] HttpClient client)
+            : base(client) {}
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
+            var userInformationEndpoint = $"{Options.UserInformationEndpoint}?access_token={tokens.AccessToken}";
+
+            if (Options.UseSignedRequests) {
+                var signature = SignRequest("/users/self",
```

Comment:
I'm not a huge fan of this hardcoded path (specially since it's essentially `userInformationEndpoint` without the scheme, domain and base path) but I can't find a better alternative, so let's keep it as-is :smile: